sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/MemberShouldBeStatic.cs

```diff
@@ -136,4 +136,15 @@ public class Class6
         [System.ObsoleteAttribute]
         public int Method2() => 0; // Compliant, any attribute disables this rule
     }
+
+    // Handle invalid code causing NullReferenceException: https://github.com/SonarSource/sonar-csharp/issues/819
+    public class Class7
+    {
+        public async Task<Result<T> Function<T>(Func<Task<Result<T>>> f)
+        {

```

Comment:
Why is methodDeclaration.Body return null here?

Reply:
Because there is a missing `>` at the end of `Task<Result<T>` so the method doesn't compile and so has neither `Body` nor ` ExpressionBody`.